core/domain/exp_services.py

```diff
@@ -1555,3 +1563,45 @@ def reject_suggestion(editor_id, thread_id, exploration_id):
             feedback_models.STATUS_CHOICES_IGNORED,
             None, 'Suggestion rejected.')
         thread.put()
+
+
+def is_draft_version_valid(exp_id, user_id):
+    """Checks if the draft version is the same as the latest version of the
+    exploration."""
+
+    draft_version = user_models.ExplorationUserDataModel.get(
+        user_id, exp_id).draft_change_list_exp_version
+    exp_version = get_exploration_summary_by_id(exp_id).version
+    return draft_version == exp_version
+
+
+def create_or_update_draft(
+        exp_id, user_id, change_list, exp_version, timestamp):
```

Comment:
s/timestamp/current_datetime

(this helps indicate that a datetime is being passed in, rather than, e.g., a value in millisecs)